### PR TITLE
Fix stats subtract

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java [hunk 1]
```diff
@@ -160,20 +160,35 @@ public StatisticRange addAndCollapseDistinctValues(StatisticRange other)
     public StatisticRange subtract(StatisticRange rightRange)
     {
         StatisticRange intersect = intersect(rightRange);
-        double newLow = getLow();
-        double newHigh = getHigh();
-        if (intersect.getLow() == getLow()) {
-            newLow = intersect.getHigh();
+
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java:163]: cmt msg "fixup:" (not "fixup!") -- will it be applied automatically?

File: presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java [hunk 2]
```diff
@@ -160,20 +160,35 @@ public StatisticRange addAndCollapseDistinctValues(StatisticRange other)
...
+        double overlapPercentOfThis = overlapPercentWith(rightRange);
+        double overlapDistinctValuesThis = overlapPercentOfThis * distinctValues;
+
+        if (intersect.getDistinctValuesCount() >= overlapDistinctValuesThis) {
+            // all NDVs from intersection are subtracted
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java:168]: What's the semantics of `subtract`? I though it was meant to be simple math, no strings attached.
If you fix it (change semantics), could you defined the new semantics?

**Review by findepi — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java (hunk 2 above)
Comment (sopel39) [reply presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java:168]: it didn't have any reasonable semantics previously. How it checks how many NDVs are removed from intersecting range and constructs result range accordingly. Sounds like an improvement to what was previously.

**Comment by sopel39** [comment]
superseded by: https://github.com/Teradata/presto/pull/684

**Pull request closed without merging**